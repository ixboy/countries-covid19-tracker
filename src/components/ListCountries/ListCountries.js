/* eslint-disable react/no-array-index-key */
import React, { useEffect } from 'react';
import CountUp from 'react-countup';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { fetchCountries } from '../../redux/worldCountries/worldCountries';
import styles from './ListCountries.module.css';

const ListCountries = () => {
  const dispatch = useDispatch();
  const allCountries = useSelector((state) => state.worldCountriesReducer.allCountries);

  useEffect(() => {
    if (!allCountries) {
      dispatch(fetchCountries());
    }
  }, [dispatch]);

  console.log(allCountries);

  if (!allCountries) {
    return 'Loading...';
  }
  return (
    <div className={styles.container}>
      {
        allCountries.map((country, i) => (
          <NavLink
            className={styles.link}
            to={`countries/${country.country}`}
            key={i}
          >
            <div className={styles.item}>
              <div className={styles.firstDiv}>
                <h2 className={styles.name}>{country.country}</h2>
                <img className={styles.image} src={country.countryInfo.flag} alt="" />
              </div>
              <div>
                <h3 type="button">Total Covid cases:</h3>
                <span className={styles.infected}>
                  <CountUp start={0} end={country.cases} duration={2.0} separator="," />
                </span>
              </div>
              <div>
                <h3 type="button">Recovered:</h3>
                <span className={styles.recovered}>
                  <CountUp start={0} end={country.recovered} duration={2.0} separator="," />
                </span>
              </div>
            </div>
          </NavLink>
        ))
      }
      ;
    </div>
  );
};

export default ListCountries;
