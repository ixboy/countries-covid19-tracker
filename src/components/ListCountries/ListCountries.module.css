.container {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.item {
	width: 300px;
	height: 200px;
  display: flex;
  flex-direction: column;
  margin: 10px;
  transition: all 0.4s;	
	border: 1px solid black;
}

.item div {
	margin: 0.4rem;
}

.item div h2{
	color: #31585a;
	font-size: 1.3rem;
}

.item div h3{
	color: #52b3b8;
}

.firstDiv {
	display: flex;
	justify-content: space-between;
}

.image {
	width: 80px;
	height: auto;
}

.infected {
  border-bottom: 1px solid rgba(0, 0, 255, 0.5);
  color: rgba(0, 0, 255, 0.5);
}

.recovered {
  border-bottom: 1px solid rgba(0, 255, 0, 0.5);
	color: rgb(25, 179, 25);
	text-align: center;
}

@media screen and (min-width: 360px) {
  .item {
    width: 160px;
		height: 160px;
  }
	.image {
		width: 40px;
		height: auto;
	}
	.item div {
		margin: 0.2rem;
		font-size: smaller;
	}

	.item *{
		font-size: 1rem;
	}
}
@media screen and (min-width: 375px) {
  .item {
    width: 175px;
		height: 160px;
  }
	.image {
		width: 40px;
		height: auto;
	}
}
@media screen and (min-width: 425px) {
  .item {
    width: 200px;
		height: 160px;
  }

	.image {
		width: 60px;
		height: auto;
	}
}

@media screen and (min-width: 780px) {
  .item {
    width: 300px;
		height: 200px;
  }
	.image {
		width: 80px;
		height: auto;
	}
}

@media screen and (min-width: 1024px) {
  .item {
    width: 275px;
  }
	.image {
		width: 120px;
		height: auto;
	}
}