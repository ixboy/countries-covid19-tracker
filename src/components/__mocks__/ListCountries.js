import React from 'react';

const allCountries = [{
  "updated": 1638671097863,
  "country": "Afghanistan",
  "countryInfo": {
    "_id": 4,
    "iso2": "AF",
    "iso3": "AFG",
    "lat": 33,
    "long": 65,
    "flag": "https://disease.sh/assets/img/flags/af.png"
  },
  "cases": 157431,
  "todayCases": 0,
  "deaths": 7312,
  "todayDeaths": 0,
  "recovered": 140617,
  "todayRecovered": 0,
  "active": 9502,
  "critical": 1124,
  "casesPerOneMillion": 3919,
  "deathsPerOneMillion": 182,
  "tests": 795556,
  "testsPerOneMillion": 19803,
  "population": 40174039,
  "continent": "Asia",
  "oneCasePerPeople": 255,
  "oneDeathPerPeople": 5494,
  "oneTestPerPeople": 50,
  "activePerOneMillion": 236.52,
  "recoveredPerOneMillion": 3500.2,
  "criticalPerOneMillion": 27.98
},
{
  "updated": 1638671097771,
  "country": "Algeria",
  "countryInfo": {
    "_id": 12,
    "iso2": "DZ",
    "iso3": "DZA",
    "lat": 28,
    "long": 3,
    "flag": "https://disease.sh/assets/img/flags/dz.png"
  },
  "cases": 211297,
  "todayCases": 0,
  "deaths": 6096,
  "todayDeaths": 0,
  "recovered": 145054,
  "todayRecovered": 0,
  "active": 60147,
  "critical": 15,
  "casesPerOneMillion": 4698,
  "deathsPerOneMillion": 136,
  "tests": 230861,
  "testsPerOneMillion": 5133,
  "population": 44976741,
  "continent": "Africa",
  "oneCasePerPeople": 213,
  "oneDeathPerPeople": 7378,
  "oneTestPerPeople": 195,
  "activePerOneMillion": 1337.29,
  "recoveredPerOneMillion": 3225.09,
  "criticalPerOneMillion": 0.33
},]

const ListCountries = () => {
  return (
    <div className="container">
      {
        allCountries.map((country, i) => (
          <div key={i}>
            <div>
              <h2 >{country.country}</h2>
              <img src={country.countryInfo.flag} alt="Flag" />
            </div>
            <div>
              <h3>Total Covid cases</h3>
              <span>
              </span>
            </div>
            <div>
              <h3>Recovered</h3>
              <span>
              </span>
            </div>
          </div>
        ))
      }
      ;
    </div>
  );
};

export default ListCountries;
