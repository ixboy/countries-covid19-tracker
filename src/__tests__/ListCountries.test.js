import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import ListCountries from '../components/__mocks__/ListCountries';
import store from '../redux/configureStore';

describe('Group of tests for countries page', () => {
  it('renders countires list correctly', () => {
    const list = renderer
      .create(<Provider store={store}><ListCountries /></Provider>)
      .toJSON();
    expect(list).toMatchSnapshot();
  });

  test('List is created', () => {
    const { container } = render(<Provider store={store}><ListCountries /></Provider>);
    const list = container.querySelector('.container');
    expect(list).toBeInTheDocument();
    expect(container).toHaveTextContent('Afghanistan');
    expect(container).toHaveTextContent('Recovered');
  });

  test('The countries Card should have the words Total, Covid, cases, etc...', () => {
    const { container } = render(<Provider store={store}><ListCountries /></Provider>);
    expect(container).toHaveTextContent('Total');
    expect(container).toHaveTextContent('Covid');
  });
});
