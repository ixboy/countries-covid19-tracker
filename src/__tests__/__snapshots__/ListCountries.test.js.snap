// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group of tests for countries page renders countires list correctly 1`] = `
<div
  className="container"
>
  <div>
    <div>
      <h2>
        Afghanistan
      </h2>
      <img
        alt="Flag"
        src="https://disease.sh/assets/img/flags/af.png"
      />
    </div>
    <div>
      <h3>
        Total Covid cases
      </h3>
      <span />
    </div>
    <div>
      <h3>
        Recovered
      </h3>
      <span />
    </div>
  </div>
  <div>
    <div>
      <h2>
        Algeria
      </h2>
      <img
        alt="Flag"
        src="https://disease.sh/assets/img/flags/dz.png"
      />
    </div>
    <div>
      <h3>
        Total Covid cases
      </h3>
      <span />
    </div>
    <div>
      <h3>
        Recovered
      </h3>
      <span />
    </div>
  </div>
  ;
</div>
`;
